### DISCUSSION TITLE
Coding on iPad without physical keyboard

### DISCUSSION THREAD

**Discussion by pedroni** [coder/code-server#4240] [General]
Hey guys, I’m loving to use code-server but it’s very difficult to code without a physical keyboard as it is very hard to hard/impossible to select code using the touchscreen.

I can easily select code with a physical keyboard using the arrow keys and command/ control/option keys but when I have only the onscreen keyboard I don’t have those keys so it’s really hard to select and copy paste, delete code. Do you guys know if we can add the arrow keys and command/ control/option keys to the iPad onscreen keyboard?

**Comment by jsjoeio** [comment]
Great question! I have no idea if this is possible. It might be worth asking in microsoft/vscode since then it would be available in code-server too.
